packages/chai-openapi-response-validator/lib/openapi-validator/lib/classes/OpenApi3Spec.js

```diff
@@ -108,4 +112,43 @@ class OpenApi3Spec extends AbstractOpenApiSpec {
   }
 }
 
+function createServerVarMap(server) {
+  if (!server.variables)
+    return new Map()
+
+  const serverVarMap = new Map();
+  const serverVars = Object.keys(server.variables);
+
+  serverVars.forEach(serverVar => {
+    serverVarMap.set(serverVar, server.variables[serverVar].enum);
+  });
+
+  return serverVarMap;
+}
+
+function replaceServerVarsInURLs (serverURLs, serverVarMap) {
```

Comment:
Although the tests prove this function works, ideally I'd prefer a simpler, more intuitive function if possible (esp without recursion). 

I'll re-review this function after we've resolved the other comments, to see if we can think of a way to write it so that a newcomer (or I) can understand it easily :)

Reply:
Happy to work on this another way, but not exactly sure how else to approach it, if you have any suggestions I'd be happy to take another crack at it.